### PR TITLE
CAMS-278 clear all filters

### PR DESCRIPTION
**Pull request by fmaddenflx**
# Purpose

Add a Clear-All button to allow the user to clear all filters.  Filter fields should visually clear and results should be displayed unfilterd.

# Major Changes

Added button and on-click to clear all fields.
Stateful values in caseDetailsScreen.tsx are cleared.
Each form component needed a clearValue() method to clear it's own state.
clearValue() methods were created for each element and added to a ref which was then exposed to the parent component.
After clearing caseDetailsScreen.tsx component state for each form element, ref.current?.clearValue() is called for each form component.

# Testing/Validation

A test was written for the case details screen component to click the button and clear all fields.

test coverage is between 93 and 94 % depending on the coverage type.

### CODE REVIEW

File: user-interface/src/case-detail/CaseDetailScreen.tsx [hunk 1]
```diff
@@ -179,18 +181,18 @@ export const CaseDetail = (props: CaseDetailProps) => {
   const [documentNumber, setDocumentNumber] = useState<number | null>(null);
   const [sortDirection, setSortDirection] = useState<SortDirection>('Newest');
   const leftNavContainerRef = useRef<CaseDetailScrollPanelRef>(null);
-
+  // const [clearAllFilters, setClearAllFilters] = useState<boolean>(false);
```
Comment (jamesobrooks) [inline user-interface/src/case-detail/CaseDetailScreen.tsx:184]: Can we remove this?

File: user-interface/src/case-detail/CaseDetailScreen.tsx [hunk 2]
```diff
@@ -240,6 +242,17 @@ export const CaseDetail = (props: CaseDetailProps) => {
     }
     setDocumentNumber(newDocumentNumber);
   }
+  function clearFilters() {
```
Comment (jamesobrooks) [inline user-interface/src/case-detail/CaseDetailScreen.tsx:245]: Nit-pick, but could we add a newline above?
```suggestion

  function clearFilters() {
```

File: user-interface/src/lib/components/MultiSelect.tsx [hunk 3]
```diff
@@ -11,9 +14,13 @@ export interface MultiSelectProps {
   closeMenuOnSelect?: boolean;
   options?: Record<string, string>[];
   label: string;
+  id: string;
 }
 
-export default function MultiSelect(props: MultiSelectProps) {
+function MultiSelectComponent(props: MultiSelectProps, ref: React.Ref<MultiSelectRef>) {
+  //const multiSelectRef =
+  //  React.useRef<SelectInstance<MultiSelectProps, true, GroupBase<MultiSelectProps>>>();
```
Comment (jamesobrooks) [inline user-interface/src/lib/components/MultiSelect.tsx:22]: Can we remove these?
```suggestion
```

File: user-interface/src/lib/components/uswds/date-range-picker.d.ts
Comment (jamesobrooks) [inline user-interface/src/lib/components/uswds/date-range-picker.d.ts:1]: From an OeSA standpoint—specifically dependency and invasive species rules—I'm thinking that interfaces like these probably should go somewhere outside the `uswds` directory. The USWDS implementation is the _less important_ code compared to the component using it. I'd be happy to discuss.

File: user-interface/src/lib/components/uswds/icon-input.d.ts
Comment (jamesobrooks) [inline user-interface/src/lib/components/uswds/icon-input.d.ts:1]: See the comment on `date-range-picker.d.ts`.

File: user-interface/src/case-detail/CaseDetailScreen.tsx (hunk 1 above)
Comment (jamesobrooks) [reply user-interface/src/case-detail/CaseDetailScreen.tsx:184]:
```suggestion
```

File: user-interface/src/lib/components/uswds/date-range-picker.d.ts [hunk 4]
```diff
@@ -0,0 +1,3 @@
+export interface DateRangePickerRef {
```
Comment (amorrow-flexion) [inline user-interface/src/lib/components/uswds/date-range-picker.d.ts:1]: Since we are doing the same thing on each input field, would it be better to move all these refs into a single file? like input-fields.d.ts. 

Unless we think we will be expanding on these individually I think we should combine them

Comment (fmaddenflx) [reply user-interface/src/lib/components/uswds/date-range-picker.d.ts:1]: More than that, we could actually use the same interface for each of the refs, if they are all the same.  Then later if we ever need something more specific for one of these components, then we can create one specific for *that* component.

File: user-interface/src/case-detail/CaseDetailScreen.tsx (hunk 2 above)
Comment (fmaddenflx) [reply user-interface/src/case-detail/CaseDetailScreen.tsx:245]: Yes I'll do that now.

File: user-interface/src/lib/components/MultiSelect.tsx (hunk 3 above)
Comment (fmaddenflx) [reply user-interface/src/lib/components/MultiSelect.tsx:22]: Yes I'll do that now.

File: user-interface/src/lib/components/uswds/date-range-picker.d.ts
Comment (fmaddenflx) [reply user-interface/src/lib/components/uswds/date-range-picker.d.ts:1]: Let's discuss

File: user-interface/src/lib/components/uswds/date-range-picker.d.ts (hunk 4 above)
Comment (fmaddenflx) [reply user-interface/src/lib/components/uswds/date-range-picker.d.ts:1]: Let's discuss

File: user-interface/src/lib/components/MultiSelect.tsx [hunk 5]
```diff
@@ -68,6 +75,21 @@ export default function MultiSelect(props: MultiSelectProps) {
     }),
   };
 
+  function clearValue() {
+    if (
+      multiSelectRef.current &&
+      Object.prototype.hasOwnProperty.call(multiSelectRef.current, 'clearValue')
```
Comment (btposey) [inline user-interface/src/lib/components/MultiSelect.tsx:81]: Since `hasOwnProperty` is on the object prototype we should be able to shorten this to `multiSelectRef.current.hasOwnProperty('clearValue')`. I believe we can also test if the clearValue function exists with a truthy expression like `&& multiSelectRef.current.clearValue`.

**Review by btposey — CHANGES_REQUESTED** [review]

**Review by btposey — APPROVED** [review]
